tests/integration/telemetry/stats/prometheus/http/customizemetrics/customize_metrics_test.go

```diff
@@ -0,0 +1,169 @@
+// +build integ
+// Copyright Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package customizemetrics
+
+import (
+	"io/ioutil"
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/prometheus"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/util/retry"
+	util "istio.io/istio/tests/integration/telemetry"
+	promUtil "istio.io/istio/tests/integration/telemetry/stats/prometheus"
+	common "istio.io/istio/tests/integration/telemetry/stats/prometheus/http"
+)
+
+var (
+	client, server echo.Instance
+	appNsInst      namespace.Instance
+	promInst       prometheus.Instance
+)
+
+func TestCustomizeMetrics(t *testing.T) {
+	framework.NewTest(t).
+		Features("observability.telemetry.stats.prometheus.customize-metric").
+		Run(func(ctx framework.TestContext) {
+			destinationQuery := buildQuery()
+			retry.UntilSuccessOrFail(t, func() error {
+				if err := sendTraffic(t); err != nil {
+					t.Errorf("failed to send traffic")
+				}
+				if err := promUtil.QueryPrometheus(t, destinationQuery, promInst); err != nil {
+					t.Logf("prometheus values for istio_requests_total: \n%s", util.PromDump(promInst, "istio_requests_total"))
+					return err
+				}
+				return nil
+			}, retry.Delay(3*time.Second), retry.Timeout(90*time.Second))
+		})
+}
+
+func TestMain(m *testing.M) {
```

Comment:
Is it possible to combine this test with existing prometheus test, so that we can avoid set up a new suite and a new prometheus.

Reply:
see the other comment, but I am open to do it either way: 1. add new test to existing stats test suite and apply envoy filter during runtime 2. create new test suite for customize metrics specifically, then we can configure that during setup with istioctl. WDYT @bianpengyuan ?

Reply:
Yes we'd still need envoy filter for attributes gen. We should exercise the code path of customizing metrics with istioctl since that is what we documented. So let's separate it out if that is easier.